idea/src/org/jetbrains/kotlin/idea/intentions/ConvertLambdaToReferenceIntention.kt

```diff
@@ -80,7 +84,11 @@ open class ConvertLambdaToReferenceIntention(textGetter: () -> String) : SelfTar
 
         val lambdaParameterIsSuspend = lambdaParameterType?.isSuspendFunctionType == true
         val calleeFunctionIsSuspend = (calleeDescriptor as? FunctionDescriptor)?.isSuspend == true
-        if (lambdaParameterIsSuspend && !calleeFunctionIsSuspend || !lambdaParameterIsSuspend && calleeFunctionIsSuspend) return false
+        if (languageVersion >= LanguageVersion.KOTLIN_1_4) {
+            if (!lambdaParameterIsSuspend && calleeFunctionIsSuspend) return false
+        } else {
+            if (lambdaParameterIsSuspend && !calleeFunctionIsSuspend || !lambdaParameterIsSuspend && calleeFunctionIsSuspend) return false
+        }
```

Comment:
This is not the correct check, since this feature (passing reference to non-suspend method as suspend function parameter) can be enabled and disabled independently from the language version. You probably should rewrite the code like this:

```suggestion
        if (!lambdaParameterIsSuspend && calleeFunctionIsSuspend) return false
        if (!languageVersionSettings.supportsFeature(LanguageFeature.SuspendConversion)) {
            if (lambdaParameterIsSuspend && !calleeFunctionIsSuspend) return false
        }
```